benchexec/result.py

```diff
@@ -133,26 +133,39 @@ class Property(collections.namedtuple("Property", "filename is_svcomp name")):
 
     __slots__ = ()  # reduce per-instance memory consumption
 
-    def compute_score(
-        self, category, result, witness_category=WITNESS_CATEGORY_UNKNOWN
-    ):
+    def _adjust_score_witness(score, witness_category):
+        if witness_category == WITNESS_CATEGORY_CORRECT:
+            # Score is already correctly set.
+            return score
+        elif witness_category == WITNESS_CATEGORY_WRONG:
+            # If a validator refutes (confirms) a wrong witness, then the score (reduction) is multiplied by a factor.
+            return score * _SCORE_FACTOR_WRONG_WITNESS
+        else:
+            # Discard result for a witness that is neither correct nor wrong.
+            return 0
+
+    def compute_score(self, category, result, witness_category=None):
         if not self.is_svcomp:
             return None
         score = _svcomp_score(category, result)
-        if witness_category == WITNESS_CATEGORY_WRONG:
-            # If a validator refutes (confirms) a wrong witness, then the score (reduction) is multiplied by a factor.
-            score *= _SCORE_FACTOR_WRONG_WITNESS
-        return score
+        if witness_category is None:
+            return score
+        else:
+            return Property._adjust_score_witness(score, witness_category)
```

Comment:
why not a regular method call?

Reply:
Because it does not access anything from self.

Reply:
I would say that doesn't matter. In Python, static methods are use quite rarely and only where necessary typically (because one does not have an instance). For helper methods called from an instance method, instance methods are fine and shorter to write.

Reply:
Done. And I added the second underscore for private methods.

Reply:
Standard Python style is to use a single underscore ("*generally, double leading underscores should be used only to avoid name conflicts with attributes in classes designed to be subclassed*" in [PEP-8](https://peps.python.org/pep-0008/)). A double underscore has a technical effect that is rarely needed and not well-known, it should be used only where required.

Reply:
Aha. Fixed.